skorch/net.py

```diff
@@ -911,10 +911,12 @@ def forward_iter(self, X, training=False, device='cpu'):
         for data in iterator:
             Xi = unpack_data(data)[0]
             yp = self.evaluation_step(Xi, training=training)
-            if isinstance(yp, tuple):
-                yield tuple(n.to(device) for n in yp)
-            else:
-                yield yp.to(device)
+            yield self._forward_output(yp, device=device)
+
+    def _forward_output(self, yp, device):
+        if isinstance(yp, tuple):
+            return tuple(n.to(device) for n in yp)
+        return yp.to(device)
```

Comment:
This is structurally very similar to `skorch.utils.to_tensor`. Maybe we should introduce a `skorch.utils.to_device` instead? This might also become handy if we support multiple GPUs in the future.

Reply:
Good point, I moved this to skorch.utils.to_device